chain/fee_manager_test.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	//"fmt"
+	"time"
+	"testing"
+
+	"github.com/ava-labs/hypersdk/math"
+
+	"go.uber.org/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnitPrice(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Mock VM
+	rules := NewMockRules(ctrl)
+	rules.EXPECT().GetMinUnitPrice().Return(Dimensions{100, 100, 100, 100, 100})
+	rules.EXPECT().GetMaxBlockUnits().Return(Dimensions{1_800_000, 2_000, 2_000, 2_000, 2_000})
+
+	// Similar to vm.go L326
+	minUnitPrice := rules.GetMinUnitPrice()
+
+	var d Dimensions
+	feeManager := NewFeeManager(nil)
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		// Set unit prices for different dimensions
+		feeManager.SetUnitPrice(i, minUnitPrice[i])
+		d[i] = uint64(i)
+	}
+
+	// Check unit prices
+	unitPrices := feeManager.UnitPrices()
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		require.Equal(unitPrices[i], feeManager.UnitPrice(i))
+	}
+
+	// Similar to spam.go
+	maxUnits := rules.GetMaxBlockUnits()
+	_, err := MulSum(unitPrices, maxUnits)
+	require.NoError(err)
+}
+
+func TestConsume(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	feeManager := NewFeeManager(nil)
+	chunks := 100
+
+	// Mock VM
+	rules := NewMockRules(ctrl)
+
+	// Similar to genesis
+	rules.EXPECT().GetMaxBlockUnits().Return(Dimensions{1_800_000, 2_000, 2_000, 2_000, 2_000})
+	rules.EXPECT().GetBaseComputeUnits().Return(uint64(1))
+	rules.EXPECT().GetStorageKeyReadUnits().Return(uint64(5))
+	rules.EXPECT().GetStorageValueReadUnits().Return(uint64(2))
+	rules.EXPECT().GetStorageKeyAllocateUnits().Return(uint64(20))
+	rules.EXPECT().GetStorageValueAllocateUnits().Return(uint64(5))
+	rules.EXPECT().GetStorageKeyWriteUnits().Return(uint64(10))
+	rules.EXPECT().GetStorageValueWriteUnits().Return(uint64(3))
+
+	// Get usage for bandwith, compute, read, allocate, writes
+	bandwidthOp := math.NewUint64Operator(200)
+	bandwidthUnits, err := bandwidthOp.Value()
+	require.NoError(err)
+
+	computeUnitsOp := math.NewUint64Operator(rules.GetBaseComputeUnits())
+	computeUnitsOp.Add(50)
+	computeUnitsOp.Add(100)
```

Comment:
Unless we are also attempting to test the math package here, I recommend just doing a single addition.